crates/agglayer-aggregator-notifier/src/sp1/mod.rs

```diff
@@ -0,0 +1,111 @@
+use std::sync::Arc;
+
+use futures::{future::BoxFuture, FutureExt as _};
+use serde::Serialize;
+use sp1_sdk::{
+    LocalProver, MockProver, NetworkProver, Prover as _, SP1ProvingKey, SP1Stdin, SP1VerifyingKey,
+};
+use tokio::task::spawn_blocking;
+
+use super::Proof;
+
+/// SP1 is a wrapper around the SP1 prover
+pub(crate) struct SP1<P> {
+    prover: Arc<P>,
+    proving_key: Arc<SP1ProvingKey>,
+    verifying_key: SP1VerifyingKey,
+}
+
+impl<P: sp1_sdk::Prover> SP1<P> {
+    pub(super) fn new(prover: P, elf: &[u8]) -> Self {
+        let (proving_key, verifying_key) = prover.setup(elf);
+
+        Self {
+            prover: Arc::new(prover),
+            proving_key: Arc::new(proving_key),
+            verifying_key,
+        }
+    }
+}
+
+impl<I> super::AggregatorProver<I> for SP1<LocalProver>
+where
+    I: Serialize,
+{
+    fn prove(&self, to_pack: Vec<I>) -> BoxFuture<'_, Result<Proof, anyhow::Error>> {
+        let mut stdin = SP1Stdin::new();
+        stdin.write(&to_pack);
```

Comment:
I see that the elf file update hasn't been reverted, is it expected that the aggregator-notifier tests remain green while it relies on the previous elf interface?

Reply:
yes the ELF is updated because of the switch of sp1 version

Reply:
Ok so the aggregator-notifier tests doesn't aim at failing in case that the proving calls fail/panic then right?

Reply:
The ELF isn't up-to-date with the current `pessimistic-proof`, it means that the tests are compatible with the ELF but not with the definition in the pessimistic-proof. The goal of the #146 is to prevent this to hapenning again